edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test/KubernetesServiceMapperTest.cs

```diff
@@ -92,47 +96,173 @@ public void UnknownProtocolDoesNotCreateService()
+            var moduleId = new ModuleIdentity("hostname", "gatewayhost", "deviceid", "moduleid", Mock.Of<ICredentials>());
 
-            var service = mapper.CreateService(moduleId, module, moduleLabels);
+            var service = mapper.CreateService(moduleId, module, moduleLabels).OrDefault();
 
-            Assert.True(service.HasValue);
+            Assert.Equal(1, service.Spec.Ports.Count);
+            AssertPort(new V1ServicePort(10, "hostport-10-tcp", null, "TCP", 10), service.Spec.Ports.First());
+            Assert.Equal("ClusterIP", service.Spec.Type);
         }
 
         [Fact]
-        public void ServiceCreationHappyPath()
+        public void PortBindingOverrideExposedPort()
         {
-            var config = new CombinedDockerConfig("image", new Docker.Models.CreateContainerParameters(), Option.None<AuthConfig>());
+            var createOptions = new Docker.Models.CreateContainerParameters
+            {
+                // Add a port to be exposed
+                ExposedPorts = new Dictionary<string, EmptyStruct>
+                {
+                    ["10/TCP"] = default(EmptyStruct)
+                },
+                // Add a port binding to host
+                HostConfig = new DockerModels.HostConfig
+                {
+                    PortBindings = new Dictionary<string, IList<DockerModels.PortBinding>>
+                    {
+                        ["10/TCP"] = new List<DockerModels.PortBinding>
+                        {
+                            new DockerModels.PortBinding
+                            {
+                                HostPort = "10"
+                            }
+                        }
+                    }
+                }
+            };
+            var config = new CombinedDockerConfig("image", createOptions, Option.None<AuthConfig>());
+            var docker = new DockerModule("module1", "v1", ModuleStatus.Running, Core.RestartPolicy.Always, Config1, ImagePullPolicy.OnCreate, DefaultConfigurationInfo, EnvVars);
+            var module = new KubernetesModule(docker, config);
+            var moduleLabels = new Dictionary<string, string>();
+            var mapper = new KubernetesServiceMapper(PortMapServiceType.ClusterIP);
             var moduleId = new ModuleIdentity("hostname", "gatewayhost", "deviceid", "moduleid", Mock.Of<ICredentials>());
-            var moduleLabels = new Dictionary<string, string> { { "label1", "value1" } };
-            config.CreateOptions.ExposedPorts = new Dictionary<string, EmptyStruct>
+
+            var service = mapper.CreateService(moduleId, module, moduleLabels).OrDefault();
+
+            Assert.Equal(1, service.Spec.Ports.Count);
+            AssertPort(new V1ServicePort(10, "hostport-10-tcp", null, "TCP", 10), service.Spec.Ports.First());
+            Assert.Equal("ClusterIP", service.Spec.Type);
+        }
+
+        [Fact]
+        public void ExposingPortsCreatesAServiceWithPorts()
```

Comment:
Nit on ordering this test before the PortBindingOverrideExposedPort for readability - since it appears that test is a prerequisite for this test :)

Reply:
Sure. Will put this change in another PR